examples/md-flexible/tests/domainDecomposition/MixedBoundaryConditionTest.cpp

```diff
@@ -0,0 +1,198 @@
+/**
+ * @file MixedBoundaryConditionTest.cpp
+ * @author S. J. Newcome
+ * @date 21/01/2022
+*/
+#include "MixedBoundaryConditionTest.h"
+#include "src/TypeDefinitions.h"
+#include "src/configuration/MDFlexConfig.h"
+#include "src/domainDecomposition/DomainTools.h"
+#include "src/domainDecomposition/RegularGridDecomposition.h"
+
+namespace {
+void initializeAutoPasContainer(RegularGridDecomposition::SharedAutoPasContainer &autoPasContainer,
+                                MDFlexConfig &configuration) {
+  autoPasContainer->setAllowedCellSizeFactors(*configuration.cellSizeFactors.value);
+  autoPasContainer->setAllowedContainers(configuration.containerOptions.value);
+  autoPasContainer->setAllowedDataLayouts(configuration.dataLayoutOptions.value);
+  autoPasContainer->setAllowedNewton3Options(configuration.newton3Options.value);
+  autoPasContainer->setAllowedTraversals(configuration.traversalOptions.value);
+  autoPasContainer->setAllowedLoadEstimators(configuration.loadEstimatorOptions.value);
+  autoPasContainer->setBoxMin(configuration.boxMin.value);
+  autoPasContainer->setBoxMax(configuration.boxMax.value);
+  autoPasContainer->setCutoff(configuration.cutoff.value);
+  autoPasContainer->setRelativeOptimumRange(configuration.relativeOptimumRange.value);
+  autoPasContainer->setMaxTuningPhasesWithoutTest(configuration.maxTuningPhasesWithoutTest.value);
+  autoPasContainer->setRelativeBlacklistRange(configuration.relativeBlacklistRange.value);
+  autoPasContainer->setEvidenceFirstPrediction(configuration.evidenceFirstPrediction.value);
+  autoPasContainer->setExtrapolationMethodOption(configuration.extrapolationMethodOption.value);
+  autoPasContainer->setNumSamples(configuration.tuningSamples.value);
+  autoPasContainer->setMaxEvidence(configuration.tuningMaxEvidence.value);
+  autoPasContainer->setSelectorStrategy(configuration.selectorStrategy.value);
+  autoPasContainer->setTuningInterval(configuration.tuningInterval.value);
+  autoPasContainer->setTuningStrategyOption(configuration.tuningStrategyOption.value);
+  autoPasContainer->setMPIStrategy(configuration.mpiStrategyOption.value);
+  autoPasContainer->setVerletClusterSize(configuration.verletClusterSize.value);
+  autoPasContainer->setVerletRebuildFrequency(configuration.verletRebuildFrequency.value);
+  autoPasContainer->setVerletSkin(configuration.verletSkinRadius.value);
+  autoPasContainer->setAcquisitionFunction(configuration.acquisitionFunctionOption.value);
+  autoPasContainer->init();
+}
+}  // namespace
+
+/**
+ * Simple test designed to show that reflection at a corner between 2 reflective faces works correctly
```

Comment:
Maybe note what the test actually does ;)
E.g. applies reflective boundaries and checks that the velocity vector flips